zulipterminal/ui_tools/boxes.py

```diff
@@ -60,36 +60,42 @@ def private_box_view(self, button: Any=None, email: str='') -> None:
     def stream_box_view(self, caption: str='', title: str='') -> None:
         self.set_editor_mode()
         self.to_write_box = None
-        self.msg_write_box = ReadlineEdit(multiline=True)
-        self.msg_write_box.enable_autocomplete(
-            func=self.generic_autocomplete,
-            key=keys_for_command('AUTOCOMPLETE').pop(),
-            key_reverse=keys_for_command('AUTOCOMPLETE_REVERSE').pop()
-        )
-        self.stream_write_box = ReadlineEdit(
-            caption="Stream:  ",
-            edit_text=caption
-        )
-        self.title_write_box = ReadlineEdit(caption="Topic:  ",
-                                            edit_text=title)
-
-        header_write_box = urwid.Columns([
-            urwid.LineBox(
-                self.stream_write_box, tlcorner='─', tline='─', lline='',
-                trcorner='┬', blcorner='─', rline='│',
-                bline='─', brcorner='┴'
-            ),
-            urwid.LineBox(
-                self.title_write_box, tlcorner='─', tline='─', lline='',
-                trcorner='─', blcorner='─', rline='',
-                bline='─', brcorner='─'
-            ),
-        ])
-        write_box = [
-            (header_write_box, self.options()),
-            (self.msg_write_box, self.options()),
-        ]
-        self.contents = write_box
+
+        if self.model.stream_dict[self.model.stream_id].get(
+                'is_announcement_only', None) == 1:
+            self.msg_footer = self.view.set_footer_text(
+                "You are not authorised to send messages to this stream.", 5)
```

Comment:
>  I'm surprised you didn't notice the interface being unresponsive after this point, until eg. you press esc.

I did notice it, but I thought it would be normal to Escape out of the warning.
But thank you for the pointer, I have fixed by moving the check below.

Reply:
Instead of doing this inside `stream_box_view` can't we add a check to `MessageBox`'s `keypress` method?

Reply:
Neil proposed 2 ways so I did it this way. I do see few `footer` warnings called in other class' `keypress()` methods. 
I will move the check as you said.